hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OzoneManager.java

```diff
@@ -1828,6 +1828,57 @@ public boolean checkAcls(ResourceType resType, StoreType storeType,
     }
   }
 
+  /**
+   * CheckAcls for the ozone object.
+   *
+   * @return true if permission granted, false if permission denied.
+   * @throws OMException ResultCodes.PERMISSION_DENIED if permission denied
+   *                     and throwOnPermissionDenied set to true.
+   */
+  @SuppressWarnings("parameternumber")
+  public boolean checkAcls(ResourceType resType, StoreType storeType,
+      ACLType aclType, String vol, String bucket, String key,
+      UserGroupInformation ugi, InetAddress remoteAddress, String hostName,
+      boolean throwIfPermissionDenied, String volumeOwner,
+      boolean recursiveAccessCheck)
+      throws OMException {
+    OzoneObj obj = OzoneObjInfo.Builder.newBuilder()
+        .setResType(resType)
+        .setStoreType(storeType)
+        .setVolumeName(vol)
+        .setBucketName(bucket)
+        .setKeyName(key).build();
+    RequestContext context = RequestContext.newBuilder()
```

Comment:
It seems every time the RequestContext changes, we will have to add a chain of checkAcls change. Can we pass a RequestContext (or its Builder) instead here for checkAcls to avoid future churns?

Reply:
> RequestContext changes, we will have to add a chain of checkAcls change

@xiaoyuyao Are you pointing to a case where user performed delete/rename on a parentDir which has millions of subpaths, now for every subpath ozoneManager needs to invoke checkAcls by creating a `RequestContext context = RequestContext.newBuilder()` ?

Please correct me if I missed something. Thanks!

Reply:
No I mean we have many checkAcls() with lots of parameters and each time, we add one more, the chain will be extended longer. Instead of doing that, we can pass the wrapper RequestContext or its Builder for checkAcls parameter.

Reply:
Thanks @xiaoyuyao for the clear explanation. I will refactor it.

Reply:
Done!